cachito/workers/tasks/general.py

```diff
@@ -49,6 +50,30 @@ def fetch_app_source(url, ref, request_id, gitsubmodule=False):
     bundle_dir = RequestBundleDir(request_id)
     log.debug("Extracting %s to %s", scm.sources_dir.archive_path, bundle_dir)
     shutil.unpack_archive(str(scm.sources_dir.archive_path), str(bundle_dir))
+    _enforce_sandbox(bundle_dir.source_root_dir)
+
+
+def _enforce_sandbox(repo_root):
+    """
+    Check that there are no symlinks that try to leave the cloned repository.
+
+    :param (str | Path) repo_root: absolute path to root of cloned repository
+    :raises ValidationError: if any symlink points outside of cloned repository
+    """
+    for dirpath, subdirs, files in os.walk(repo_root):
+        dirpath = Path(dirpath)
+
+        for entry in subdirs + files:
+            full_path = dirpath / entry
+            real_path = full_path.resolve()
+            try:
+                real_path.relative_to(repo_root)
+            except ValueError:
+                # This one cannot fail
```

Comment:
Right, because we're not using the real path here. Can we improve the comment so readers don't have to think as hard about this? :wink:

Reply:
Good point, fixed :smile: